### PR TITLE
:sparkles: drivers

### PR DESCRIPTION
**Pull request by fabienjuif**
Welcome to drivers :)

This is inspired from cycleJS drivers.

The purpose is 
 - to have easy to test `reactions` (inject some side effects handlers like `fetch`)
 - to share some behaviour between reactions

Example of basic usage:
```js

const fetchConfig = reaction(async (action, store, { http }) => {
  const config = await http('http://ma-config.com')
  store.config.set(config)
}) 

const store = createStore({
  config: simpleObject(), 
}, {
  listeners: [
    when('@@kst/INIT')(fetchConfig),
  ],
})
```

Example of usage (should be integrated to the lib in next PR):
```js
// this PR doesn't allow myFetch to access store :( next one, but this is the idea
const myFetch = store => async (...args) => {
  store.dispatch('HTTP_LOADING')
  try {
     const data = await fetch(...args)
  } catch (ex) {
    store.dispatch({ type: 'HTTP_ERROR', payload: ex }) 
    return
  }
  store.dispatch({ type: 'HTTP_OK', payload: data })
}

const fetchConfig = reaction(async (action, store, { http }) => {
  const config = await http('http://ma-config.com') // dispatch all events
  store.config.set(config)
}) 

const store = createStore({
  config: simpleObject(), 
}, {
  listeners: [
    when('@@kst/INIT')(fetchConfig),
  ],
  drivers: {
    http: myFetch
  },
})
```

### CODE REVIEW

File: src/createStore/createStore.js [hunk 1]
```diff
@@ -11,11 +11,21 @@ const defaultOptions = {
   listeners: undefined,
   devtools: true,
   name: 'store',
+  drivers: {
+    http: (...args) => fetch(...args),
```
Comment (EmrysMyrddin) [inline src/createStore/createStore.js]: Why do you define a new function instead of passing directly `fetch` ?

Comment (guillaumecrespel) [reply src/createStore/createStore.js]: so you can overload it and have a standardized name.

File: src/createStore/createStore.js [hunk 2]
```diff
@@ -11,11 +11,21 @@ const defaultOptions = {
...
+  },
 }
 
 export default (definition, options = defaultOptions) => {
```
Comment (guillaumecrespel) [inline src/createStore/createStore.js]: `options = defaultOptions` is useless

File: src/createStore/createStore.js (hunk 1 above)
Comment (fabienjuif) [reply src/createStore/createStore.js]: I can do `http: fetch` yes.
 - But in the lib I want to have something that dispatch classics events from http (STARTED/LOADING/ENDED/ERROR)

File: src/createStore/createStore.js (hunk 2 above)
Comment (fabienjuif) [reply src/createStore/createStore.js]: This is not uselesss since I use options.drivers

**Comment by fabienjuif** [comment]
I will not fix that decrease in coverage for now

**Pull request merged**